### PR TITLE
bugfix/19926-annotations-incorrect-event-binding

### LINKED ISSUE #19926: Text label rotation using `useHTML`  causing issues with control points.
Creating new issue for https://github.com/highcharts/highcharts/issues/19861#issuecomment-1757847822.

Uncommenting `useHTML` seems to resolve Issue 1 and Issue 2, but I want to rotate the text label in annotations. Can someone suggest alternative / workaround here? Thanks.

### PR DESCRIPTION
**Pull request by bm64**
Fixed #19926, click event wasn't bound correctly for annotations with HTML labels.

### CODE REVIEW

**Review by kamil-musialowski — APPROVED** [review]

File: samples/unit-tests/annotations/annotations-events/demo.js [hunk 1]
```diff
@@ -305,4 +305,42 @@ QUnit.test('Annotations events - general', function (assert) {
...
+                y: 100
+            }],
+            type: 'path',
+            strokeWidth: 10
+        }],
+        draggable: false
+    });
+
+    controller.click(chart.plotLeft + 105, chart.plotTop + 100);
+    controller.click(chart.plotLeft + 200, chart.plotTop + 75);
```
Comment (pawelfus) [inline samples/unit-tests/annotations/annotations-events/demo.js:339]: Instead of hardcoded pixels, please translate x & y values to pixels

**Review by pawelfus — CHANGES_REQUESTED** [review]

File: samples/unit-tests/annotations/annotations-events/demo.js (hunk 1 above)
Comment (pawelfus) [reply samples/unit-tests/annotations/annotations-events/demo.js:339]: Just realized annotation uses pixels position too, so it is fine

**Review by pawelfus — APPROVED** [review]
Thanks!

**Review by pawellysy — APPROVED** [review]

**Review by DJTechnoo — APPROVED** [review]
Looks great!

**Review by TorsteinHonsi — APPROVED** [review]

**Pull request merged**